File: Arrays/Plus One/Python /reverse_traversal.py
def plusOne(digits):
    """
    Adds one to the number represented by digits array
    using reverse traversal.
    """
    n = len(digits)

    for i in range(n - 1, -1, -1):
        if digits[i] < 9:
            digits[i] += 1
            return digits
        digits[i] = 0

    # If all digits were 9
    return [1] + digits


# Example usage
if __name__ == "__main__":
    print(plusOne([1, 2, 3]))   # [1, 2, 4]
    print(plusOne([9, 9, 9]))   # [1, 0, 0, 0]

File: Arrays/Plus One/Cpp/reverse_traversal.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> plusOne(vector<int>& digits) {
    int n = digits.size();

    for (int i = n - 1; i >= 0; i--) {
        if (digits[i] < 9) {
            digits[i]++;
            return digits;
        }
        digits[i] = 0;
    }

    // If all digits were 9
    digits.insert(digits.begin(), 1);
    return digits;
}

int main() {
    vector<int> digits = {9, 9, 9};
    vector<int> result = plusOne(digits);

    for (int d : result)
        cout << d << " ";

    return 0;
}
